### ISSUE TITLE
Reload content on filesystem change / trigger reload manually

### ISSUE THREAD

**Issue by brunopbarbosa** [nuxt/content#783]
Hello All,
 
I have been going through the documentation for a few weeks without any success.

We have a scenario where the 'content' folder in production is updated separately from the build (it's fed from a different repo) so that people maintaining the markdown docs don't have access to the build files.

Trouble is that whenever that content is updated, the pods need to be restarted in order for new files / updated files to be available to the frontend.

Is there any way to trigger the reload of the content folder automatically, either by reading changes on the filesystem or as a workaround, as a manual trigger?

Thanks in advance, 
Bruno

**Comment by NozomuIkuta** [comment]
These issues may be related to what you want: 

- #265 (How get content from other repos ?)
- #736 (Fech data from an external app)

**Comment by brunopbarbosa** [comment]
> These issues may be related to what you want:
> 
> * #265 (How get content from other repos ?)
> * #736 (Fech data from an external app)

Hi, thanks! Although these are more related to getting the content, which I am already doing successfully.

My issue is reloading it into the frontend without having to rebuild the app.

**Comment by brunopbarbosa** [comment]
Hi @Atinux! 

Yes, that's my case exactly.

**Comment by brunopbarbosa** [comment]
Oh so at the moment there is no way of doing that, right? 
At least I can't stop racking my brain trying to figure out how to do it 😅
I will keep an eye on it, thanks for letting me know 👍

**Comment by sShrek1** [comment]
Still no update on this? I am getting data from external service, add json file to my /content/ folder and then nothing?
Are there any work-arounds? 

Is Content not supposed to work with SSR?

**Comment by sShrek1** [comment]
@Tahul Yes! My first understanding with nuxt/content was that we can develop and deploy to our prod-environment and do changes, that wasn't the case each change needed a new deploy. The files should be updated and the content cache should be repopulated/rebuilt. 
Out of box we should keep it simple as it is but there should be native function/trigger when you want to rebuild the cache. Having a watcher on each file would be heavy, in my case I have 300 files atm and will be around 900 each year. 

Just by having this will help developers create their own simple CMS. 
How cool wouldn't it be just building one simple application with only nuxt and nothing else?! :D

**Comment by sShrek1** [comment]
Here is an example from gatsby https://www.gatsbyjs.com/docs/refreshing-content/

**Comment by johan-Rm** [comment]
> 

Hi @Tahul,

I'm very interested in hot reloading data when the source is changed, in my case a .json file ('/content/*.json') using the @nuxt-content module.

Do you have any news ? I just redid a test this morning it does not appear in the updates yet

**Comment by ENBYSS** [comment]
> Hey there :)
> 
> Multi-source will be supported natively from @nuxt/content v2.
> 
> I think the behavior you are talking about (updating content when remote source changed) should be supported by default, but I'll keep this open as I think it's not yet the case.
> 
> We are close of having multi-source ready to handle multiple kind of sources (remote / fs), but I think filesystem ones should be handled by default.
> 
> My only question is knowing if the content cache will get populated with these new datas at production time, but I'm leaving this answer to @farnabaz.
> 
> Thanks for opening this issue, expect updates soon.

Checking since it's been 7 months and Content V2 is already out - are there any updates on this?

**Comment by zenatshockbyte** [comment]
Nuxt studio (https://nuxt.studio/docs/studio/live-preview#how-does-it-work), which is linked at the top of the nuxt-content website, created by nuxt-labs now supports a 'live-preview' in production mode like asked for in this thread.

I would honestly not hold your breath for this getting to nuxt-content itself at this point but fingers crossed

Though you might just be able to use nuxt-studio yourself instead so hopefully that'll help :)

**Comment by hedongshu** [comment]
I have the same requirement as you. My markdown files are all saved in GitHub, and I hope that nuxt-content can automatically update after I update the markdown files in GitHub.

**Comment by brunopbarbosa** [comment]
> > I have the same requirement as you. My markdown files are all saved in GitHub, and I hope that nuxt-content can automatically update after I update the markdown files in GitHub.
> 
> By using a pipeline, you can trigger a (static) build of your nuxt project each time you upload a file in Github.
> 
> However, if your project runs on a nuxt server and you want to update your prod in realtime, without having to trigger a build, I suggest using nuxt MDC (https://github.com/nuxt-modules/mdc) instead of nuxt content.

As far as i'm aware, MDC is used with nuxt content too - how do you suggest to use MDC instead of Nuxt Content for live content updating?

**Comment by LoganTann** [comment]
Afaik I would store articles in my own database, and use mdc to convert markdown to vue components. So you wouldn't be using Nuxt content at all, as if you were building your own CMS.

**Comment by brunopbarbosa** [comment]
OK, thanks, that does seem like an idea worth investigating!
*Bruno Barbosa*

+ 351 913 346 869
+48 690 983 269
https://bbarbosa.dev

**Comment by brunopbarbosa** [comment]
@farnabaz , just for clarification, is there any movement on this or should we assumed it won't happen as it's a feature available on Nuxt Studio? No harm done if it's the case, would just like to know if it's something we can expect to have or not.

**Comment by alex-eri** [comment]
> using nuxt MDC

Needs to implement navigaton, search, prerender etc...

**Comment by oneminch** [comment]
I don't think it's yet possible to update production content in a separate repo and see it reflected in your app without a rebuild or a new deployment. Content is not *fetched* at runtime from the remote source. It's _fetched and bundled_ with your site at build time. 

The indexing and database generation happen during the build phase, which means the application relies on the static dump generated at that time. Even if content is edited with the new Nuxt Studio module, the site needs to be rebuilt and redeployed for the changes to be reflected.

**Comment by oneminch** [comment]
Hi @brunopbarbosa, I discovered this library called Comark Content (https://content.comark.dev/). It's a framework-agnostic solution from the same people behind Nuxt Content but solves the exact problem you are trying to solve. It's still in its early days but if this is a problem you still need to solve, I recommend you keep an eye on it :)
